README.md

```diff
@@ -1,71 +1,196 @@
+
+### Speech Pipeline
+
+The Speech Pipeline is the module that ties together VAD (voice activity detection), wakeword, and ASR (automated speech detection). The VAD listens to a frame of audio captured by the input device to determine if speech is present. If it is, the wakeword model processes subsequent frames of audio looking for the keyword it has been trained to recognize. If the keyword is found, the pipeline is activated and performs speech recognition, converting the subsequent audio into a transcript. The Speech Pipeline is initialized like this:
+
+```python
+from spokestack.io.pyaudio import PyAudioInput
+from spokestack.pipeline import SpeechPipeline
+from spokestack.vad.webrtc import VoiceActivityDetector
+from spokestack.wakeword.tflite import WakewordTrigger
+from spokestack.asr.spokestack.speech_recognizer import SpeechRecognizer
+
+mic = PyAudioInput()
+vad = VoiceActivityDetector()
+wake = WakewordTrigger("path_to_tflite_model")
+asr = SpeechRecognizer("spokestack_id", "spokestack_secret")
+
 
-### TFLite Interpreter
+pipeline = SpeechPipeline(mic, [vad, wake, asr])
+pipeline.start()
+pipeline.run()
+```
 
-In addition to the Python dependencies, you will need to install the TFLite Interpreter. You can install it for your platform by following the instructions at [TFLite Interpreter](https://www.tensorflow.org/lite/guide/python#install_just_the_tensorflow_lite_interpreter).
-**Note:** this is not the full [Tensorflow](https://www.tensorflow.org/) package.
+Now that the pipeline is running, it becomes important to access the results from processes at certain events. For example, when speech is recognized there is a `recognize` event. These events allow code to be executed outside the pipeline in response. The process of registering a response is done with a pipeline callback, which we will cover in the next section.
+
+#### Pipeline Callbacks
+
+Pipeline callbacks allow additional code to be executed when a speech event is detected. For example, we can print when the pipeline is activated by registering a function with the `pipeline.event` decorator.
+
+```python
+@pipeline.event
+def on_activate(context):
+    print(context.is_active)
+```
+
+One of the most important use cases for a pipeline callback is accessing the ASR transcript for additional processing by the NLU. The transcript is accessed with the following:
+
+```python
+@pipeline.event
+def on_recognize(context):
+    print(context.transcript)
+```
+
+### Natural Language Understanding (NLU)
+
+Natural Language Understanding turns an utterance into a machine readable format. For our purposes, this is joint intent detection and slot filling. You can read more about the concepts [here](https://www.spokestack.io/docs/concepts/nlu). We like to think of intents as the action a user desires from an application, and slots as the optional arguments to fulfill the requested action. Our NLU model is initialized like this:
```

Comment:
```suggestion
Natural Language Understanding turns an utterance into structured data a machine can act on. For our purposes, this is joint intent detection and slot filling. You can read more about the concepts [here](https://www.spokestack.io/docs/concepts/nlu). We like to think of intents as the action a user desires from an application, and slots as the optional arguments to fulfill the requested action. Our NLU model is initialized like this:
```

The fact that it's so hard to define this in a way that makes it sound like a useful addition should be a red flag to the community that maybe their current approach doesn't add that much value...